src/databricks/labs/ucx/workspace_access/clusters.py

```diff
@@ -0,0 +1,117 @@
+import logging
+
+from databricks.labs.blueprint.installation import Installation
+from databricks.labs.blueprint.tui import Prompts
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import InvalidParameterValue
+from databricks.sdk.service.compute import ClusterDetails, DataSecurityMode
+
+logger = logging.getLogger(__name__)
+
+
+class ClusterAccess:
+    def __init__(self, installation: Installation, ws: WorkspaceClient, prompts: Prompts):
+        self._ws = ws
+        self._prompts = prompts
+        self._installation = installation
+
+    def list_cluster(self):
+        cluster_list = {}
+        clusters = self._ws.clusters.list()
+        for cluster in clusters:
+            if cluster.cluster_source is not None and cluster.cluster_source.name != "JOB":
+                cluster_list[cluster.cluster_name] = cluster.cluster_id
+        return cluster_list
+
+    def _get_cluster_id(self, cluster_id: str):
+        if cluster_id != "<ALL>":
+            return [x.strip() for x in cluster_id.split(",")]
+        cluster_list = []
+        clusters = self._ws.clusters.list()
```

Comment:
Using the Request parameter to perform remote filtering of results, reducing network, speeding api call `w.clusters.list(compute.ListClustersRequest())`
Eliminating some logic below

Reply:
#DRY - dont repeat yourself

Doesn't make sense to have two independent calls to the` _ws.clusters.list()` API, for one to 'project' IDs and the other to bring back the entire object.
One should call the other, and do list comprehension to return the list id properties.

Reply:
Now the API call is happening only once. filtering the cluster using the list comprehension